api/report_server.thrift

```diff
@@ -303,9 +308,45 @@ service codeCheckerDBAccess {
   // Analysis result storage related API calls.
   //============================================
 
-  // store checker run related data to the database
-  // by default updates the results if name was already found
-  // using the force flag removes existing analysis results for a run
+  // !!!WARNING!!! The store API has a very strict way of usage! The best if
+  // one calls the storeZip() function which handles the storage of a full run.
+  // See the documentation of the storeZip() function.
+  // The other storage functions can also be used to store reports. The calling
+  // order must be:
+  // 1. addFileContent()
+  // 2. addCheckerRun()
+  // 3. addReport(), addSkipPath(), setRunDuration()
+  // 4. finishCheckerRun()
+
+  // The client can ask the server whether a file is already stored in the
+  // database. If it is, then it is not necessary to send it in the ZIP file
+  // with storeZip() function. This function requires a list of file hashes
+  // (sha256) and returns the ones which are not stored yet.
+  list<string> necessaryFileContents(
+                                     1: list<string> file_hashes)
```

Comment:
We use multiple naming conventions in the thrift file:
 - **CamelCase variables** (reportId, reportFilters, requestError, etc.)
 - **Underscore variables** (file_hashes, base_run_id, diff_type, etc.)

We have to use only one convention.